nucypher/blockchain/eth/agents.py

```diff
@@ -518,6 +518,21 @@ def bond_operator(self, staking_provider: ChecksumAddress, operator: ChecksumAdd
         return receipt
 
 
+def _process_active_staking_providers_info(
+    active_staking_providers_info: Iterable,
+) -> Tuple[types.TuNits, Dict[ChecksumAddress, types.TuNits]]:
+    total_authorized_tokens, staking_providers_info = active_staking_providers_info
+    staking_providers = dict()
+    for info in staking_providers_info:
+        staking_provider_address = to_checksum_address(info[0:20])
+        staking_provider_authorized_tokens = to_int(info[20:32])
+        staking_providers[staking_provider_address] = types.TuNits(
+            staking_provider_authorized_tokens
+        )
+
+    return types.TuNits(total_authorized_tokens), staking_providers
```

Comment:
maybe this could be done directly in `get_active_staking_providers`? wdyt?

Reply:
I thought about it but my only concern was changing the API since `get_active_staking_providers` is a public method for `TACoApplicationAgent`. Although thinking about it now, the API changed anyway since whatever was returned before is not compatible with what is returned now anyway.

👍 I'll update it.

Reply:
✔️